packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -122,7 +122,23 @@ abstract class FlutterCommand extends Command<Null> {
     _usesPubOption = true;
   }
 
-  void addBuildModeFlags({ bool defaultToRelease: true }) {
+  void usesBuildNumberOption() {
+    argParser.addOption('buildNumber',
+        help: 'An integer used as an internal version number.\n'
```

Comment:
It's an integer on Android. It's a x.y.z string on iOS (with optional y and z).

Reply:
To have a consistent version number only an int is accepted. I added a type check in `getBuildInfo()`